docs/plugins/dry-run-testing.md

```diff
@@ -0,0 +1,134 @@
+
+```typescript
+export function exampleAction() {
+  return createTemplateAction<{
+    example: string;
+  }>({
+    id: 'action:example',
+    description: 'Example action',
+    schema: {
+      input: {
+        type: 'object',
+        properties: {
+          example: {
+            title: 'example',
+            type: 'string',
+          },
+        },
+      },
+    },
+    supportsDryRun: true,
+    async handler(ctx) {
+      ...
+    },
+  });
+}
+```
+
+## Adding handling for dry run
+
+To add handling for dry run functionality you need to add a check for 'ctx.isDryRun' inside the handler of the configuration object which is being passed into 'createTemplateAction' in the function where the behavior of your action is defined. Once the check is successful, you can perform the desired actions expected in a dry run, e.g. outputting non-sensitive inputs.
+
+```typescript
+async handler(ctx) {
+      ...
+
+      // If this is a dry run, log and return
+      if (ctx.isDryRun) {
+        ctx.logger.info(`Dry run complete`);
+        return;
+      }
+
+      ...
+    },
+```
+
+## Testing dry run handling
+
+You will also need to add tests for the dry run handling, for example:
+
+```typescript
+  it('should not perform action during dry run', async () => {
+    ...
+
+    // Create the context object with the necessary properties for a dry run
+    const ctx = {
+      ...mockContext,
+      isDryRun: true,
+      input: {
+        ...
+      },
+    };
+
+    // Call the handler with the context
+    await action.handler(ctx);
+
+    expect(...);
+  });
+```
+
+## Dry run via API call
+
+Dry run can be performed using the dry-run API, which allows the dry run to be completed in code.
+This [command line script](https://github.com/backstage/backstage/blob/master/contrib/scaffolder/template-testing-dry-run.md) offers a way for you to do work with dry-run API. You run it against a running instance, either locally or remote, and use it like:
+
+```bash
+scaffolder-dry http://localhost:7007/ template-directory values.yml output-directory
+```
+
+If you're using backend permissions, pass a front-end auth token from a current browser session via --token $FRONTEND_TOKEN.
+
+You can also query the dry run endpoint in code, for example:
+
+`dry-run.js`
+
+```javascript
+const template = yaml.load(
+  await fs.readFile('path/to/templates/template.yaml', 'utf-8'),
+);
+const values = JSON.parse(
+  await fs.readFile('path/to/templates/template_values.json', 'utf-8'),
+);
+
+// Prepare the request body
+const body = {
+  template,
+  values,
+  directoryContents,
+};
+
+// Send the request to the dry-run endpoint
+const response = await fetch(
+  'http://localhost:7000/api/scaffolder/v2/dry-run',
+  {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: 'Bearer <YOUR_API_KEY>', // replace 'YOUR_API_KEY' with your actual API key
```

Comment:
There are many API Keys in the Backstage world, we need to be more clear about which one.